bootstrap/eks/controllers/eksconfig_controller.go

```diff
@@ -312,7 +312,7 @@ func (r *EKSConfigReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Man
 	}
 
 	err = c.Watch(
-		&source.Kind{Type: &clusterv1.Cluster{}},
+		source.Kind(mgr.GetCache(), &clusterv1.Cluster{}),
```

Comment:
unit tests are failing due to this, passing a cache to source.Kind does not seem to be properly working and tests are failing due to the object is missing, looking into it

Reply:
Thanks @richardcase for helping to identify the issue with fakeclient and fixing them!